docs/user-guide/rules/http-compression.md

```diff
@@ -0,0 +1,567 @@
+# Require resources to be served compressed (`http-compression`)
+
+`http-compression` warns against not serving resources compressed when
+requested as such using the most appropriate encoding.
+
+## Why is this important?
+
+One of the fastest and easiest ways one can improve the web site's/app's
+performance is to reduce the amount of data that needs to get delivered
+to the client by using HTTP compression. This not only [reduces the data
+used by the user][wdmsc], but can also significallty cut down on the
+server costs.
+
+However, there are a few things that need to be done right in order for
+get the most out of compression:
+
+* Only compress resources for which the result of the compression
+  will be smaller than original size.
+
+  In general text-based resources (HTML, CSS, JavaScript, SVGs, etc.)
+  compresss very well especially if the file is not very small.
+  The same goes for some other file formats (e.g.: ICO files, web fonts
+  such as EOT, OTF, and TTF, etc.)
+
+  However, compressing resources that are already compressed (e.g.:
+  images, audio files, PDFs, etc.) not only waste CPU resources, but
+  usually result in little to no reduction, or in some cases even
+  a bigger file size.
+
+  The same goes for resources that are very small because of the
+  overhead of compression file formats.
+
+* Use most efficien compression method.
+
+  `gzip` is the most used encoding nowadays as it strikes a good
+  balance between compression ratio (as high as 70-90% especially
+  for larger files) and encoding time, and is supported pretty much
+  eveywere.
+
+  Better savings can be achieved using [`Zopfli`][zopfli] which
+  can reduce the size on average with [3–8% more then `gzip`][[zopfli
+  blog post]. Since `Zopfli` output (for the `gzip` option) is valid
+  `gzip` content, `Zopfli` works eveywere `gzip` works. The only
+  downsize is that encoding takes way more time than with `gzip`,
+  thus, making `Zopfli` more suitable for static content (i.e. encoding
+  resources with is as part of build script, not on the fly).
+
+  But, things can be improved even futher using [Brotli][brotli]. This
+  new encoding allows to get [20–26% higher compression ratios][brotli
```

Comment:
Maybe remove `new`?  Don't know when we will review this docs and it's been out for some time already.

Reply:
Done.